src/backend/core/resource_server/views.py

```diff
@@ -0,0 +1,54 @@
+"""Resource Server views"""
+
+from joserfc.jwk import KeySet
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from .auth import ResourceServerAuthentication
+from .utils import import_private_key_from_settings
+
+
+class JWKSView(APIView):
+    """
+    API endpoint for retrieving a JSON Web Key (JWK).
+
+    Returns:
+        Response: JSON response containing the JWK data.
+    """
+
+    authentication_classes = []  # disable authentication
+    permission_classes = []  # disable permission
+
+    def get(self, request):
+        """Handle GET requests to retrieve the JSON Web Key (JWK).
+
+        Returns:
+            Response: JSON response containing the JWK data.
+        """
+
+        private_key = import_private_key_from_settings()
+        jwk = KeySet([private_key]).as_dict()
```

Comment:
are we exposing our private key here and if yes shouldn't we expose the public key instead?

Reply:
I have already verified (but will do an extra check). The public Key should be computed by the `KeySet` constructor from the private key. No public key was exposed here.